distributed/tests/test_scheduler.py

```diff
@@ -225,6 +225,22 @@ def test_remove_worker_by_name_from_scheduler(s, a, b):
     s.validate_state()
 
 
+@gen_cluster(client=True, Worker=Nanny)
+def test_remove_worker_increments_suspicious(client, s, a, b):
+    def killer(x):
+        import sys
+
+        sys.exit()
+
+    futures = client.map(killer, range(20))
+    while len(s.workers) == 2:
+        yield gen.sleep(0.01)
+    assert sum(tp.suspicious for tp in s.task_prefixes.values()) == sum(
+        ts.suspicious for ts in s.tasks.values()
+    )
+    assert 10 <= s.task_prefixes["killer"].suspicious <= 30
```

Comment:
@mrocklin I changed the test to not depend on internal APIs, but I'm not really happy with the way the test works right now. For me it was flapping, as sometimes one worker is killed until the wait is over and sometimes two are killed which leads to either 10 tasks marked as suspicious or 30.
Do you have an idea how to solve this properly? I tried to await the submitted tasks, but then I got `KilledWorker` exceptions.

Reply:
Maybe something like the following?

```python
@gen_cluster(client=True, Worker=Nanny, config={"distributed.scheduler.suspicious-tasks": 1})  # TODO verify config key
async def test_suspicious_prefix(client, s, a, b):
    future = client.submit(sys.exit, 0)
    await wait(future)
    assert s.task_prefixes["exit"].suspicious
```

This isn't as strong of a test as what you have here (there is only one task), but it should be simpler and run more more quickly.

Reply:
It especially takes away the flapping, because of the `allowed-failures` configuration. Thanks a lot for the help @mrocklin